includes/class-woopay-tracker.php

```diff
@@ -217,6 +217,17 @@ public function is_country_tracks_eligible() {
 	 * @return bool
 	 */
 	public function should_enable_tracking( $is_admin_event = false, $track_on_all_stores = false ) {
+		// Allow merchants to disable all shopper tracking via filter.
+		if ( ! apply_filters( 'wcpay_shopper_tracking_enabled', true ) ) {
+			return false;
+		}
+
+		// Respect WooCommerce global tracking opt-out setting.
+		// Only disable if explicitly set to 'no' (not false, null, or empty).
+		$allow_tracking = get_option( 'woocommerce_allow_tracking', '' );
+		if ( 'no' === $allow_tracking ) {
+			return false;
+		}
```

Comment:
```suggestion
		// Allow merchants to disable all shopper tracking via filter.
		if ( ! apply_filters( 'wcpay_shopper_tracking_enabled', 'no' !== get_option( 'woocommerce_allow_tracking', '' ) ) ) {
			return false;
		}
```

I think maybe also this can use the same filter structure?

Reply:
@frosso my idea was to create precedence between the filter and the option, allowing the filter to override the option value. Let me know if you feel this is too extra.